### ISSUE TITLE
phpPackages.imagick fails to build: wand/MagickWand.h: No such file or directory

### ISSUE THREAD

**Issue by league** [NixOS/nixpkgs#28727]
## Issue description

After a recent channel update, I'm getting a build failure in `phpPackages.imagick`. Using `git blame`, the definition of `phpPackages.imagick` hasn't changed recently, but the imagemagick definition itself has been updated in a391aca (I don't know for certain that it's related).

### Steps to reproduce

```
$ nix-shell -p phpPackages.imagick
these derivations will be built:
  /nix/store/iaalgglw43q79n38zadd99lrglgkgdgl-php-imagick-3.4.3RC1.drv
building path(s) ‘/nix/store/8129dp9hgf09pcz1590sr7gdb2q8n1br-php-imagick-3.4.3RC1’
unpacking sources
unpacking source archive /nix/store/p5rm8zzhj2pkgxwywlxd11rg85fifb2n-imagick-3.4.3RC1.tgz
source root is imagick-3.4.3RC1
setting SOURCE_DATE_EPOCH to timestamp 1465502516 of file imagick-3.4.3RC1/tests/skipprobefourier.inc
patching sources
autoreconfPhase
Configuring for:
PHP Api Version:         20160303
Zend Module Api No:      20160303
Zend Extension Api No:   320160303
configuring
fixing libtool script ./ltmain.sh
configure flags: --disable-static --prefix=/nix/store/8129dp9hgf09pcz1590sr7gdb2q8n1br-php-imagick-3.4.3RC1 --with-imagick=/nix/store/16pfm5rjwr8b3grax2b0d1rkp30l63pc-imagemagick-6.9.9-7-dev 
checking for grep that handles long lines and -e... /nix/store/r6s8rcd28wsk4gwviyc93343bq5zwlqq-gnugrep-3.0/bin/grep
checking for egrep... /nix/store/r6s8rcd28wsk4gwviyc93343bq5zwlqq-gnugrep-3.0/bin/grep -E
checking for a sed that does not truncate output... /nix/store/xk38vw7z7bfr8173vdwfrfamxqcaj7hi-gnused-4.4/bin/sed
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking how to run the C preprocessor... gcc -E
checking for icc... no
checking for suncc... no
checking whether gcc and cc understand -c and -o together... yes
checking for system library directory... lib
checking if compiler supports -R... no
checking if compiler supports -Wl,-rpath,... yes
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking target system type... x86_64-unknown-linux-gnu
checking for PHP prefix... /nix/store/dqygaqfid5mx95nmy4xnp4ml2607g0m2-php-7.1.7
checking for PHP includes... -I/nix/store/dqygaqfid5mx95nmy4xnp4ml2607g0m2-php-7.1.7/include/php -I/nix/store/dqygaqfid5mx95nmy4xnp4ml2607g0m2-php-7.1.7/include/php/main -I/nix/store/dqygaqfid5mx95nmy4xnp4ml2607g0m2-php-7.1.7/include/php/TSRM -I/nix/store/dqygaqfid5mx95nmy4xnp4ml2607g0m2-php-7.1.7/include/php/Zend -I/nix/store/dqygaqfid5mx95nmy4xnp4ml2607g0m2-php-7.1.7/include/php/ext -I/nix/store/dqygaqfid5mx95nmy4xnp4ml2607g0m2-php-7.1.7/include/php/ext/date/lib
checking for PHP extension directory... $(out)/lib/php/extensions
checking for PHP installed headers prefix... /nix/store/dqygaqfid5mx95nmy4xnp4ml2607g0m2-php-7.1.7/include/php
checking if debug is enabled... no
checking if zts is enabled... yes
checking for re2c... no
configure: WARNING: You will need re2c 0.13.4 or later if you want to regenerate PHP parsers.
checking for gawk... gawk
checking whether to enable the imagick extension... yes, shared
checking for pkg-config... /nix/store/clvw27qshazq458waarrjqj0k5qpnjkn-pkg-config-0.29/bin/pkg-config
checking ImageMagick MagickWand API configuration program... checking Testing /nix/store/16pfm5rjwr8b3grax2b0d1rkp30l63pc-imagemagick-6.9.9-7-dev/bin/MagickWand-config... It exists
found in /nix/store/16pfm5rjwr8b3grax2b0d1rkp30l63pc-imagemagick-6.9.9-7-dev/bin/MagickWand-config
checking if ImageMagick version is at least 6.2.4... found version 6.9.9 Q16 
checking for MagickWand.h or magick-wand.h header... user location /nix/store/16pfm5rjwr8b3grax2b0d1rkp30l63pc-imagemagick-6.9.9-7-dev/include/ImageMagick-6/wand/MagickWand.h
/nix/store/16pfm5rjwr8b3grax2b0d1rkp30l63pc-imagemagick-6.9.9-7-dev/bin/MagickWand-config: line 53: /nix/store/clvw27qshazq458waarrjqj0k5qpnjkn-PKG_CONFIG_PATH=/nix/store/16pfm5rjwr8b3grax2b0d1rkp30l63pc-imagemagick-6.9.9-7-dev/lib/pkgconfig: No such file or directory
/nix/store/16pfm5rjwr8b3grax2b0d1rkp30l63pc-imagemagick-6.9.9-7-dev/bin/MagickWand-config: line 41: /nix/store/clvw27qshazq458waarrjqj0k5qpnjkn-PKG_CONFIG_PATH=/nix/store/16pfm5rjwr8b3grax2b0d1rkp30l63pc-imagemagick-6.9.9-7-dev/lib/pkgconfig: No such file or directory
checking PHP version is at least 5.1.3... yes. found 7.1.7
libs



checking for MagickGetVersion... no
checking for __MagickGetVersion... no
checking for MagickSetImageInterpolateMethod... no
checking for __MagickSetImageInterpolateMethod... no
checking for ld used by gcc... /nix/store/j1yn1np5g2vgm5yp8qmqg67x8riwy2kr-gcc-wrapper-5.4.0/bin/ld
checking if the linker (/nix/store/j1yn1np5g2vgm5yp8qmqg67x8riwy2kr-gcc-wrapper-5.4.0/bin/ld) is GNU ld... yes
checking for /nix/store/j1yn1np5g2vgm5yp8qmqg67x8riwy2kr-gcc-wrapper-5.4.0/bin/ld option to reload object files... -r
checking for BSD-compatible nm... /nix/store/3iggy4d1wdwivs4kx9ic90npiwxvnrr8-binutils-2.27/bin/nm -B
checking whether ln -s works... yes
checking how to recognize dependent libraries... pass_all
./configure: line 5491: /usr/bin/file: No such file or directory
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking dlfcn.h usability... yes
checking dlfcn.h presence... yes
checking for dlfcn.h... yes
checking the maximum length of command line arguments... 1572864
checking command to parse /nix/store/3iggy4d1wdwivs4kx9ic90npiwxvnrr8-binutils-2.27/bin/nm -B output from gcc object... ok
checking for objdir... .libs
checking for ar... ar
checking for ranlib... ranlib
checking for strip... strip
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC
checking if gcc PIC flag -fPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking whether the gcc linker (/nix/store/j1yn1np5g2vgm5yp8qmqg67x8riwy2kr-gcc-wrapper-5.4.0/bin/ld) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no

creating libtool
appending configuration tag "CXX" to libtool
checking whether the g++ linker (/nix/store/j1yn1np5g2vgm5yp8qmqg67x8riwy2kr-gcc-wrapper-5.4.0/bin/ld) supports shared libraries... yes
checking for g++ option to produce PIC... 
checking if g++ static flag  works... yes
checking if g++ supports -c -o file.o... yes
checking whether the g++ linker (/nix/store/j1yn1np5g2vgm5yp8qmqg67x8riwy2kr-gcc-wrapper-5.4.0/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... GNU/Linux ld.so
(cached) (cached) checking how to hardcode library paths into programs... unsupported
configure: creating ./config.status
config.status: creating config.h
building
make flags: SHELL=/nix/store/lpk84rsbha199vm3k54498lqv2jswqj8-bash-4.4-p5/bin/bash EXTENSION_DIR=$(out)/lib/php/extensions   
/nix/store/lpk84rsbha199vm3k54498lqv2jswqj8-bash-4.4-p5/bin/bash /tmp/nix-build-php-imagick-3.4.3RC1.drv-0/imagick-3.4.3RC1/libtool --mode=compile gcc  -I. -I/tmp/nix-build-php-imagick-3.4.3RC1.drv-0/imagick-3.4.3RC1 -DPHP_ATOM_INC -I/tmp/nix-build-php-imagick-3.4.3RC1.drv-0/imagick-3.4.3RC1/include -I/tmp/nix-build-php-imagick-3.4.3RC1.drv-0/imagick-3.4.3RC1/main -I/tmp/nix-build-php-imagick-3.4.3RC1.drv-0/imagick-3.4.3RC1 -I/nix/store/dqygaqfid5mx95nmy4xnp4ml2607g0m2-php-7.1.7/include/php -I/nix/store/dqygaqfid5mx95nmy4xnp4ml2607g0m2-php-7.1.7/include/php/main -I/nix/store/dqygaqfid5mx95nmy4xnp4ml2607g0m2-php-7.1.7/include/php/TSRM -I/nix/store/dqygaqfid5mx95nmy4xnp4ml2607g0m2-php-7.1.7/include/php/Zend -I/nix/store/dqygaqfid5mx95nmy4xnp4ml2607g0m2-php-7.1.7/include/php/ext -I/nix/store/dqygaqfid5mx95nmy4xnp4ml2607g0m2-php-7.1.7/include/php/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /tmp/nix-build-php-imagick-3.4.3RC1.drv-0/imagick-3.4.3RC1/imagick_file.c -o imagick_file.lo 
mkdir .libs
 gcc -I. -I/tmp/nix-build-php-imagick-3.4.3RC1.drv-0/imagick-3.4.3RC1 -DPHP_ATOM_INC -I/tmp/nix-build-php-imagick-3.4.3RC1.drv-0/imagick-3.4.3RC1/include -I/tmp/nix-build-php-imagick-3.4.3RC1.drv-0/imagick-3.4.3RC1/main -I/tmp/nix-build-php-imagick-3.4.3RC1.drv-0/imagick-3.4.3RC1 -I/nix/store/dqygaqfid5mx95nmy4xnp4ml2607g0m2-php-7.1.7/include/php -I/nix/store/dqygaqfid5mx95nmy4xnp4ml2607g0m2-php-7.1.7/include/php/main -I/nix/store/dqygaqfid5mx95nmy4xnp4ml2607g0m2-php-7.1.7/include/php/TSRM -I/nix/store/dqygaqfid5mx95nmy4xnp4ml2607g0m2-php-7.1.7/include/php/Zend -I/nix/store/dqygaqfid5mx95nmy4xnp4ml2607g0m2-php-7.1.7/include/php/ext -I/nix/store/dqygaqfid5mx95nmy4xnp4ml2607g0m2-php-7.1.7/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /tmp/nix-build-php-imagick-3.4.3RC1.drv-0/imagick-3.4.3RC1/imagick_file.c  -fPIC -DPIC -o .libs/imagick_file.o
In file included from /tmp/nix-build-php-imagick-3.4.3RC1.drv-0/imagick-3.4.3RC1/php_imagick_file.h:24:0,
                 from /tmp/nix-build-php-imagick-3.4.3RC1.drv-0/imagick-3.4.3RC1/imagick_file.c:22:
/tmp/nix-build-php-imagick-3.4.3RC1.drv-0/imagick-3.4.3RC1/php_imagick_defs.h:29:31: fatal error: wand/MagickWand.h: No such file or directory
compilation terminated.
make: *** [Makefile:194: imagick_file.lo] Error 1
builder for ‘/nix/store/iaalgglw43q79n38zadd99lrglgkgdgl-php-imagick-3.4.3RC1.drv’ failed with exit code 2
error: build of ‘/nix/store/iaalgglw43q79n38zadd99lrglgkgdgl-php-imagick-3.4.3RC1.drv’ failed
/run/current-system/sw/bin/nix-shell: failed to build all dependencies
```

I verified that `wand/MagickWand.h` does indeed exist, if I find it this way:

```
$ nix-shell -p imagemagick.dev pkgconfig
[nix-shell:~/tmp/nixpkgs]$ pkg-config --cflags ImageMagick
-fopenmp -DMAGICKCORE_HDRI_ENABLE=0 -DMAGICKCORE_QUANTUM_DEPTH=16 -I/nix/store/16pfm5rjwr8b3grax2b0d1rkp30l63pc-imagemagick-6.9.9-7-dev/include/ImageMagick-6

[nix-shell:~/tmp/nixpkgs]$ head /nix/store/16pfm5rjwr8b3grax2b0d1rkp30l63pc-imagemagick-6.9.9-7-dev/include/ImageMagick-6/wand/MagickWand.h
/*
  Copyright 1999-2017 ImageMagick Studio LLC, a non-profit organization
  dedicated to making software imaging solutions freely available.

  You may not use this file except in compliance with the License.
  obtain a copy of the License at

    https://www.imagemagick.org/script/license.php

  Unless required by applicable law or agreed to in writing, software

[nix-shell:~/tmp/nixpkgs]$
```

and now that I'm looking more closely at the log (above), I see that the configure reported that it found it too, but there's something broken there, related to the pkgconfig spec:

```
checking ImageMagick MagickWand API configuration program... checking Testing /nix/store/16pfm5rjwr8b3grax2b0d1rkp30l63pc-imagemagick-6.9.9-7-dev/bin/MagickWand-config... It exists
found in /nix/store/16pfm5rjwr8b3grax2b0d1rkp30l63pc-imagemagick-6.9.9-7-dev/bin/MagickWand-config
checking if ImageMagick version is at least 6.2.4... found version 6.9.9 Q16 
checking for MagickWand.h or magick-wand.h header... user location /nix/store/16pfm5rjwr8b3grax2b0d1rkp30l63pc-imagemagick-6.9.9-7-dev/include/ImageMagick-6/wand/MagickWand.h
/nix/store/16pfm5rjwr8b3grax2b0d1rkp30l63pc-imagemagick-6.9.9-7-dev/bin/MagickWand-config: line 53: /nix/store/clvw27qshazq458waarrjqj0k5qpnjkn-PKG_CONFIG_PATH=/nix/store/16pfm5rjwr8b3grax2b0d1rkp30l63pc-imagemagick-6.9.9-7-dev/lib/pkgconfig: No such file or directory
/nix/store/16pfm5rjwr8b3grax2b0d1rkp30l63pc-imagemagick-6.9.9-7-dev/bin/MagickWand-config: line 41: /nix/store/clvw27qshazq458waarrjqj0k5qpnjkn-PKG_CONFIG_PATH=/nix/store/16pfm5rjwr8b3grax2b0d1rkp30l63pc-imagemagick-6.9.9-7-dev/lib/pkgconfig: No such file or directory
```

## Technical details

* System: NixOS, currently 17.03.1700.51a83266d1 (but trying to update when I encountered this)
* Nix version: nix-env (Nix) 1.11.13
* Nixpkgs version: 17.03.1775.56da88a298
* Sandboxing enabled: false

**Comment by vbgl** [comment]
I think it has been addressed there: 82a4c5e38b8aaa9787eb184adaa54f06b6f68f8e.

**Comment by league** [comment]
@vbgl  Indeed, that solves it. So this is a plea to cherry-pick 82a4c5e into the stable nixos-17.03 channel, but otherwise I can close it.
